chromium_src/third_party/blink/renderer/core/origin_trials/origin_trial_context.cc

```diff
@@ -0,0 +1,53 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "third_party/blink/renderer/core/origin_trials/origin_trial_context.h"
+
+#define AddFeature AddFeature_ChromiumImpl
+#define AddForceEnabledTrials AddForceEnabledTrials_ChromiumImpl
+#include "../../../../../../../third_party/blink/renderer/core/origin_trials/origin_trial_context.cc"
+#undef AddForceEnabledTrials
+#undef AddFeature
+
+namespace blink {
+
+// AddFeature doesn't check if origin_trials::IsTrialValid.
+void OriginTrialContext::AddFeature(OriginTrialFeature feature) {
+  // When updating this list also update the trials names below in
+  // AddForceEnabledTrials override as well as the names in
+  //  * origin_trials.cc override and
+  static const std::array<OriginTrialFeature, 3> kBraveDisabledTrialFeatures = {
+      // [cr88] OriginTrialFeature::kDigitalGoods,
+      OriginTrialFeature::kNativeFileSystem,
+      OriginTrialFeature::kSignedExchangeSubresourcePrefetch,
+      OriginTrialFeature::kSubresourceWebBundles,
+  };
+  if (base::Contains(kBraveDisabledTrialFeatures, feature))
+    return;
+
+  AddFeature_ChromiumImpl(feature);
+}
+
+// AddForceEnabledTrials only has a DCHECK with origin_trials::IsTrialValid.
+void OriginTrialContext::AddForceEnabledTrials(
+    const Vector<String>& trial_names) {
+  // When updating this list also update
+  //  * origin_trials.cc override and
+  static const char* const kBraveDisabledTrialNames[] = {
+      // [cr88] "DigitalGoods",
+      "NativeFileSystem2",
```

Comment:
Can we remove duplication with https://github.com/brave/brave-core/pull/6933/files#diff-dc54666ac5547a38442ebd8a02187f6df83b883532bb933408fa753a3c214868R24?

Reply:
Good point, I will rework this a bit.